src/main/java/com/scottlogic/TopicFinder.java

```diff
@@ -0,0 +1,33 @@
+package com.scottlogic;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class TopicFinder {
+
+    List<String> stopWords = new ArrayList<>();
+
+    public TopicFinder(List<String> stopWords) {
+        this.stopWords = stopWords;
+    }
+
+    public String topicFinder(UserPost inputPost) {
+
+        String result = "";
+        if (inputPost != null) {
+            String content = inputPost.getContents();
+            String[] allWords = content.replaceAll("[^a-zA-Z ]", "").toLowerCase().split(" ");
+            StringBuilder builder = new StringBuilder();
+            for (String word : allWords) {
+                if (!stopWords.contains(word)) {
```

Comment:
This just finds all the topics mentioned in the post, then in Topic sort we select main topic based on topics present, and if all are repeated the same number of times, the first topic is assigned as the main topic of that post

Reply:
Are we looking at this part of the instructions? If we are, it looks like a topic should be something used more than once.
> Write a new class which will extract topics from Post content using regex or otherwise (unit tests should be agnostic to the implementation here). Here we are considering a topic as a keyword that is used multiple times.